### ISSUE TITLE
FR: map overlay support

### ISSUE THREAD

**Issue by sikmir** [tumic0/GPXSee#178]
Are there any plans for map overlay support? It's often needed to combine some base map (e.g. OSM) and some specific/local map (e.g. orienteering, mountains, OpenRailwayMap, etc).

**Comment by tumic0** [comment]
Not at the moment. It would require a complete rewrite of the maps related code to make it universal (with reprojecting), even only a limited overlay support for maps of the same coordinates system would be a lot of work. Although GPXSee supports a broad range of maps, this is IMHO something that already belongs to some GIS rather than into a simple GPS log viewer. But lets let it open.

**Comment by MeisterP** [comment]
This feature would be handy to add hillshading tiles...

See https://wiki.openstreetmap.org/wiki/Hike_%26_Bike_Map#Tiles for example urls

**Comment by scls19fr** [comment]
It will be a nice feature! But I understand it can be complex to implement.

**Comment by tumic0** [comment]
Since a958544667179b8c6b207fc79724eea5ef9f28fa there is at least support for GARMIN IMG maps overlays.

If the map name is `mapname.img` and there is also a file named `mapname.img.1` (.2, .3, ... for additional layers) in the same path, than it will be automatically loaded as an overlay of `mapname.img`.

**Comment by sikmir** [comment]
> Since [a958544](https://github.com/tumic0/GPXSee/commit/a958544667179b8c6b207fc79724eea5ef9f28fa) there is at least support for GARMIN IMG maps overlays.

Does GPXSee support raster maps in GARMIN IMG format?

**Comment by tumic0** [comment]
> Does GPXSee support raster maps in GARMIN IMG format?

No. Such maps are not very common (maybe except marine charts) and the format is not known. Spending time on reverse engineering this part of IMG maps is IMHO not worth it.

If someone wants to get his brain damaged by reverse engineering the biggest mess in software engineering history then the NT routable links (aka roads in NT maps) are a much better way to do so...

**Comment by sikmir** [comment]
> No. Such maps are not very common (maybe except marine charts) and the format is not known. Spending time on reverse engineering this part of IMG maps is IMHO not worth it.

Just in case:
* [SAS.Planet](http://www.sasgis.org/wikisasiya/doku.php/%D1%8D%D0%BA%D1%81%D0%BF%D0%BE%D1%80%D1%82_%D0%BA%D1%8D%D1%88%D0%B0_%D0%B2_%D1%84%D0%BE%D1%80%D0%BC%D0%B0%D1%82_.img_%D0%B4%D0%BB%D1%8F_%D0%BD%D0%B0%D0%B2%D0%B8%D0%B3%D0%B0%D1%82%D0%BE%D1%80%D0%BE%D0%B2_garmin) supports exports to raster IMG
* Reverse engineering of raster IMG: http://whiter.brinkster.net/raster_img.shtml

**Comment by sikmir** [comment]
> Since [a958544](https://github.com/tumic0/GPXSee/commit/a958544667179b8c6b207fc79724eea5ef9f28fa) there is at least support for GARMIN IMG maps overlays.
> 
> If the map name is `mapname.img` and there is also a file named `mapname.img.1` (.2, .3, ... for additional layers) in the same path, than it will be automatically loaded as an overlay of `mapname.img`.

I've tried the following:
1. Download `OV-Hib-Lov-20191028-1.02.img` map from [here](https://drive.google.com/file/d/10aAOKY8U7TQvFvuWBkwTei9iP3-cUvoE/view?usp=sharing) into `~/.local/share/gpxsee/maps/test.img.1`
2. Create IMG map from GPX file ([file](https://www.wikiloc.com/back-country-skiing-trails/202003-lovozero-48144243), [IMGfromGPX tool](https://www.javawa.nl/imgfromgpx_en.html)) and place it into `~/.local/share/gpxsee/maps/test.img`
3. Result:
![scr0](https://user-images.githubusercontent.com/688044/80427952-76cf1480-88f1-11ea-92c8-dcd38d6ba21e.jpg)

That's almost perfect at least for my use case (show all my tracks over some basemap without loading all tracks as gpx)! Thanks!

**Comment by tumic0** [comment]
Online maps overlay support added in version 15.4. So much for the overlays topic, any advanced cross-map-type overlays will not happen, things are already complicated enough...

**Comment by eddy-geek** [comment]
for the record,
fixed in https://github.com/tumic0/GPXSee/commit/8254d39bf45e7821184d10b0317cd8732f12281b
documentation of the feature in https://www.gpxsee.org/doc.html :

> Since version 15.4 multiple URLs can be given. The resulting map is then composed of overlayed images, one layer for every URL present.

```
<?xml version="1.0" encoding="UTF-8"?>
<map xmlns="http://www.gpxsee.org/map/1.7">
  <name>Open Sea Map</name>
  <url>https://t2.openseamap.org/tile/$z/$x/$y.png</url>
  <url>https://tiles.openseamap.org/seamark/$z/$x/$y.png</url>
</map>
```

**Comment by eddy-geek** [comment]
On my side I've kept using the work I started in https://github.com/tumic0/GPXSee/pull/491 despite its very raw state. (Although more bugs have crept up following some hasty rebase.)

For my use-case I don't mind that the layer support is only single-map-type, but I need offline/MBTiles support.
Now it seems feasible to setup eg [mbtileserver](https://github.com/consbio/mbtileserver) on top of my MBTiles and put `<url>https://localhost/services/file1/tile/$z/$x/$y.png</url>` but that's a bit cumbersome 🥲 

it would be awesome if I could put `<url>./file1.mbt</url>` (where the path can be relative to xml location or absolute) and GPXSee did the job ;)
